skyportal/handlers/api/internal/source_views.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+from sqlalchemy import func, desc
+from baselayer.app.access import permissions, auth_or_token
+from ...base import BaseHandler
+from ....models import (
+    DBSession, Source, SourceView, GroupSource
+)
+
+
+class SourceViewsHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        prefs = (self.current_user.preferences if
+                 hasattr(self.current_user, 'preferences') and
+                 self.current_user.preferences else {})
```

Comment:
Is this a check for None?  Should then be: "and self.current_user.preferences is not None.

But cleaner may be:

`getattr(self.current_user, 'preferences') or {}`

Reply:
Agreed, thanks